src/passes/OptimizeInstructions.cpp

```diff
@@ -1255,6 +1265,8 @@ struct OptimizeInstructions
   Expression* optimizeWithConstantOnRight(Binary* curr) {
     using namespace Match;
     Builder builder(*getModule());
+    Const* c;
+    Binary* inner;
```

Comment:
It would be good to move these variables down to where they are used and introduce a new scope because these are not common to nearly all the patterns in this function like `left`, `right` and `type` are.

```
{
  Const* c;
  Binary* inner;
  ... code from below ...
}
```